### PR TITLE
Issue LIBCLOUD-455: validate image id

### PR DESCRIPTION
**Pull request by virtuald**
https://issues.apache.org/jira/browse/LIBCLOUD-455

### CODE REVIEW

File: libcloud/compute/drivers/openstack.py [hunk 1]
```diff
@@ -1875,7 +1881,7 @@ def _to_node(self, api_node):
                 # Docs says "tenantId", but actual is "tenant_id". *sigh*
                 # Best handle both.
                 tenantId=api_node.get('tenant_id') or api_node['tenantId'],
-                imageId=api_node['image']['id'],
+                imageId=image_id,
```
Comment (Kami) [inline libcloud/compute/drivers/openstack.py]: More robust and Pythonic way would be to just do `imageId=api_node.get('image', {}).get('id', None)`

**Comment by Kami** [comment]
It would also be good to use raw response data (https://gist.github.com/virtuald/1358cec79751925e3817) as a test fixture and add a test case for this specific scenario.

**Comment by virtuald** [comment]
@Kami I'd add a test fixture, but it's not exactly clear how to do that, and I don't have the time to figure it out right now.

**Comment by Kami** [comment]
@virtuald I've added a regression test case, modified the code to correctly handle the missing attribute and merged changes into trunk.

The approach I have originally suggested in my comment was wrong (sorry about that!). I thought that the `image` attribute value was an empty `dict`, but it was actually an empty string (nasty API).

This pull request can be closed now.

Thanks!

**Comment by virtuald** [comment]
Thanks!

**Pull request closed without merging**